report_xlsx/static/src/js/report/qwebactionmanager.js

```diff
@@ -0,0 +1,48 @@
+// © 2017 Creu Blanca
+// License AGPL-3.0 or later (https://www.gnuorg/licenses/agpl.html).
+odoo.define('report_xlsx.report', function(require){
+'use strict';
+
+var ActionManager= require('web.ActionManager');
+var crash_manager = require('web.crash_manager');
+var framework = require('web.framework');
+
+ActionManager.include({
+    _executeReportAction: function (action, options) {
+        var self = this;
+        var cloned_action = _.clone(action);
+        if (cloned_action.report_type === 'xlsx') {
+            framework.blockUI();
+            var report_xlsx_url = 'report/xlsx/' + cloned_action.report_name;
+            if (_.isUndefined(cloned_action.data) ||
+                _.isNull(cloned_action.data) ||
+                (_.isObject(cloned_action.data) && _.isEmpty(cloned_action.data)))
+            {
+                if(cloned_action.context.active_ids) {
+                    report_xlsx_url += '/' + cloned_action.context.active_ids.join(',');
+                }
+            } else {
+                report_xlsx_url += '?options=' + encodeURIComponent(JSON.stringify(cloned_action.data));
+                report_xlsx_url += '&context=' + encodeURIComponent(JSON.stringify(cloned_action.context));
+            }
+            self.getSession().get_file({
+                url: report_xlsx_url,
+                data: {data: JSON.stringify([
+                    report_xlsx_url,
+                    cloned_action.report_type
+                ])},
+                error: crash_manager.rpc_error.bind(crash_manager),
+                success: function (){
+                    if(cloned_action && options && !cloned_action.dialog){
+                        options.on_close();
+                    }
+                }
+            });
+            framework.unblockUI();
+            var closeAction = { type: 'ir.actions.act_window_close' };
+            return self.doAction(closeAction, _.pick(options, 'on_close'));
+        }
+        return self._super(action, options);
+    }
+});
+});
```

Comment:
line

Reply:
Hi @elicoidal , done.

Reply:
Changes done. Travis and Runbot test passed. 
Thanks.